File: RepoParticipant.java
package me.repository;

import me.model.Participant;
import me.model.Proba;
import me.model.Stiluri;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class RepoParticipant implements IRepoParticipant {
    private static final Logger logger = LogManager.getLogger();
    JdbcUtils jdbcUtils;

    public RepoParticipant(Properties props){
        logger.info("Initializing RepoAngajat with properties: {} ",props);
        jdbcUtils=new JdbcUtils(props);
    }

    @Override
    public List<Participant> findParticipantiDupaNume(String nume) {
        Connection con=jdbcUtils.getConnection();
        List<Participant> participanti = new ArrayList<>();
        try(PreparedStatement preStmt=con.prepareStatement("select * from Participanti where nume=?")){
            preStmt.setString(1,nume);
            try(ResultSet result=preStmt.executeQuery()) {
                 while (result.next()) {
                    int id = result.getInt("id");
                    int varsta = result.getInt("varsta");
                    participanti.add(new Participant(id,nume,varsta));
                }
                 return participanti;
            }
        }catch (SQLException ex){
            System.out.println("Error DB "+ex);
        }
        return null;
    }

    @Override
    public List<Participant> findParticipantiDupaVarsta(int i) {
        return null;
    }

    @Override
    public Participant save(Participant elem) {
        logger.info("Se conecteaza la baza de date");
        Connection con=jdbcUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("insert into Participanti (nume,varsta) values (?,?)")){
            preStmt.setString(1,elem.getNume());
            preStmt.setInt(2,elem.getVarsta());
            preStmt.executeUpdate();
        }catch (SQLException ex){
            logger.info("eroare baza de date: "+ex.getMessage());
        }
        logger.info("S-a adaugat in baza de date participant "+elem);
        return findParticipantiDupaNume(elem.getNume()).get(0);
    }

    @Override
    public Participant update(Participant elem) { return null; }

    @Override
    public Participant delete(Participant elem) throws IllegalArgumentException {
        return null;
    }

    @Override
    public Participant get(Integer integer) {
        Connection con=jdbcUtils.getConnection();
        List<Participant> participanti = new ArrayList<>();
        try(PreparedStatement preStmt=con.prepareStatement("select * from Participanti where id=?")){
            preStmt.setInt(1,integer);
            try(ResultSet result=preStmt.executeQuery()) {
                if (result.next()) {
                    int id = result.getInt("id");
                    String nume  = result.getString("nume");
                    int varsta = result.getInt("varsta");
                    return new Participant(id,nume,varsta);
                }
            }
        }catch (SQLException ex){
            System.out.println("Error DB "+ex);
        }
        return null;
    }

    @Override
    public int size() { return 0; }

    @Override
    public Iterable<Participant> findAll() {
        Connection con=jdbcUtils.getConnection();
        List<Participant> participanti = new ArrayList<>();
        try(PreparedStatement preStmt=con.prepareStatement("select * from Participanti")){
            try(ResultSet result=preStmt.executeQuery()) {
                while (result.next()) {
                    int id = result.getInt("id");
                    String nume = result.getString("nume");
                    int varsta = result.getInt("varsta");
                    participanti.add(new Participant(id,nume,varsta));
                }
                return participanti;
            }
        }catch (SQLException ex){
            System.out.println("Error DB "+ex);
        }
        return null;
    }
}

File: Concurs Inot C#/RepoParticipant.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Problema_3_CSharp.me.model;
using log4net;

namespace Problema_3_CSharp.me.repository
{
    public class RepoParticipant : IRepoParticipant
    {
        private static readonly ILog log = LogManager.GetLogger("LoggerRepoParticipant");
        public Participant Delete(Participant elem)
        {
            throw new NotImplementedException();
        }

        public IEnumerable<Participant> FindAll()
        {
            throw new NotImplementedException();
        }

        public IEnumerable<Participant> FindParticipantiDupaNume(string nume)
        {
            log.Info("Se incepe conectarea la bd si selectarea participantilor dupa numele " + nume);
            var con = DBUtils.getConnection();
            IList<Participant> participanti = new List<Participant>();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from Participanti where nume=@nume";
                var paramNume = comm.CreateParameter();
                paramNume.ParameterName = "@nume";
                paramNume.Value = nume;
                comm.Parameters.Add(paramNume);

                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        int id = dataR.GetInt32(0);
                        int varsta = dataR.GetInt32(2);
                        Participant p = new Participant(id, nume, varsta);
                        participanti.Add(p);
                    }
                }
            }
            return participanti;
        }

        public IEnumerable<Participant> FindParticipantiDupaVarsta(int i)
        {
            throw new NotImplementedException();
        }

        public Participant Get(int id)
        {
            log.InfoFormat("Se cauta participantul cu id-ul {0}", id);
            var con = DBUtils.getConnection();

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from Participanti where id=@id";
                var paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                comm.Parameters.Add(paramId);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        String nume = dataR.GetString(1);
                        int varsta = dataR.GetInt32(2);
                        Participant p = new Participant(id, nume, varsta);
                        log.InfoFormat("S-a gasit {0}", p);
                        return p;
                    }
                }
            }
            log.InfoFormat("Nu s-a gasit niciun participant cu id-ul {0}", id);
            return null;
        }


        public Participant Save(Participant elem)
        {
            log.Info("Se incearca adaugarea");
            var con = DBUtils.getConnection();
            using (var comm = con.CreateCommand())
            {

                comm.CommandText = "insert into Participanti (nume,varsta)  values (@nume, @varsta)";
                var paramNume = comm.CreateParameter();
                paramNume.ParameterName = "@nume";
                paramNume.Value = elem.Nume;
                comm.Parameters.Add(paramNume);

                var paramVarsta = comm.CreateParameter();
                paramVarsta.ParameterName = "@varsta";
                paramVarsta.Value = elem.Varsta;
                comm.Parameters.Add(paramVarsta);

                var result = comm.ExecuteNonQuery();

                log.Info("S-a adaugat " + elem.ToString());
                return FindParticipantiDupaNume(elem.Nume).First();
            }
        }

        public int Size()
        {
            throw new NotImplementedException();
        }

        public Participant Update(Participant elem)
        {
            throw new NotImplementedException();
        }
    }
}
